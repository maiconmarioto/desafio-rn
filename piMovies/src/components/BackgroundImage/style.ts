import styled from 'styled-components/native';
import {Dimensions} from 'react-native';

const width = Dimensions.get('screen').width;

export const ImageBG = styled.ImageBackground`
  justify-content: center;
  align-items: center;
  height: 200px;
  background-color: #000;
`;

export const Container = styled.View`
  flex: 1;
  height: 200px;
  background-color: rgba(0, 0, 0, 0.5);
  flex-grow: 1;
  flex-basis: 100%;
  width: ${width / 2 - 2}px;
`;
