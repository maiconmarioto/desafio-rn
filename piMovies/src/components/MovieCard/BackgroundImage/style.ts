import styled from 'styled-components/native';

export const ImageBG = styled.ImageBackground`
  flex: 1;
  justify-content: center;
`;

export const Container = styled.View`
  flex: 1;
  border-radius: 20px;
  background-color: rgba(0, 0, 0, 0.5);
`;
