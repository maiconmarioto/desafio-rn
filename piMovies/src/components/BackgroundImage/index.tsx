import React from 'react';

import {ImageBG} from './style';

interface Props {
  src: string;
  opacity?: number;
}

const BackgroundImage: React.FC<Props> = ({src, opacity = 0.5, children}) => {
  return (
    <ImageBG source={{uri: src}} resizeMode="cover" imageStyle={{opacity}}>
      {children}
    </ImageBG>
  );
};

export default BackgroundImage;
