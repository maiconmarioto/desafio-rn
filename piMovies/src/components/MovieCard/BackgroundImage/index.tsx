import React from 'react';
import {ImageBackgroundProperties} from 'react-native';

import {Container, ImageBG} from './style';

const BackgroundImage: React.FC<ImageBackgroundProperties> = ({
  source,
  borderRadius,
  children,
}) => {
  return (
    <ImageBG imageStyle={{borderRadius}} source={source} resizeMode="cover">
      <Container>{children}</Container>
    </ImageBG>
  );
};

export default BackgroundImage;
