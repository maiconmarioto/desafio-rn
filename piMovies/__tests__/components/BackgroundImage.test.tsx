import React from 'react';
import renderer from 'react-test-renderer';

import BackgroundImage from '../../src/components/BackgroundImage';

test('renders correctly', () => {
  const alert = renderer
    .create(
      <BackgroundImage opacity={1} src="https://via.placeholder.com/350x65" />,
    )
    .toJSON();
  expect(alert).toMatchSnapshot();
});
