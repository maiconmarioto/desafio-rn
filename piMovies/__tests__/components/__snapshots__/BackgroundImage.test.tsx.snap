// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#000",
        "height": 200,
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "https://via.placeholder.com/350x65",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        Object {
          "opacity": 1,
        },
      ]
    }
  />
</View>
`;
